### PR TITLE
[stable/external-dns] Add connection specifics for Cloudflare

### PR DESCRIPTION
**Pull request by compleatang**
My work is based off of #1859 and assumes that it will be merged prior to review and possible merger of this work. 

- reference: https://github.com/kubernetes-incubator/external-dns/blob/master/docs/tutorials/cloudflare.md

### CODE REVIEW

**Comment by compleatang** [comment]
I have signed the CLA.

File: stable/external-dns/templates/deployment.yaml [hunk 1]
```diff
@@ -32,6 +32,16 @@ spec:
           {{- range $key, $value := .Values.extraArgs }}
             - --{{ $key }}={{ $value }}
           {{- end }}
+          env:
+          {{- if and .Values.cloudflare.api_key .Values.cloudflare.email }}
```
Comment (mgoodness) [inline stable/external-dns/templates/deployment.yaml]: Nit: indent the `if` & `end` statements one level less than the block they wrap.

File: stable/external-dns/templates/deployment.yaml [hunk 2]
```diff
@@ -32,6 +32,16 @@ spec:
...
+          - name: CF_API_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ template "external-dns.fullname" . }}
+                key: cloudflare_api_key
+          - name: CF_API_EMAIL
+            value: {{ .Values.cloudflare.email }}
```
Comment (mgoodness) [inline stable/external-dns/templates/deployment.yaml]: Wrap templates in quotes if they simply echo user-provided strings (e.g. `value: "{{ .Values.cloudflare.email }}"`).

File: stable/external-dns/templates/secret.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{- if and .Values.cloudflare.api_key .Values.cloudflare.email -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels: {{ include "external-dns.labels" . | indent 4 }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: {{ template "external-dns.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (mgoodness) [inline stable/external-dns/templates/secret.yaml]: Don't add `namespace` field to manifests; Tiller does that for us.

**Review by mgoodness — CHANGES_REQUESTED** [review]
Just a few formatting/convention nits.

**Comment by compleatang** [comment]
Thanks for the nits; always good to learn convention! Everything is updated.

**Review by mgoodness — APPROVED** [review]

**Pull request merged**